src/rust/rust-proto-new/src/protocol/status.rs

```diff
@@ -0,0 +1,363 @@
+use tonic::{Code as GrpcCode, Status as GrpcStatus};
+
+
+/// Status codes used by [`Status`].
+///
+/// These variants closely mirror the [gRPC status codes].
+///
+/// [gRPC status codes]: https://github.com/grpc/grpc/blob/master/doc/statuscodes.md#status-codes-and-their-use-in-grpc
```

Comment:
Do we actually need to support all these codes? For example, where will we use a `DeadlineExceeded` or a `FailedPrecondition`? Let's start out supporting only the error states we intend to actually use.

Reply:
The only two I can imagine removing are DeadlineExceeded and, perhaps, DataLoss. FailedPrecondition is indeed something I could imagine returning - for example, if a service tried to access a table in a database but the table had not been created, just as a somewhat contrived example.

So I could remove DeadlineExceeded and DataLoss?

Reply:
Well, I think instead it would be best to let requirements drive the design here. Which status variants do we currently have a demonstrated (e.g. in a service) need to support? Let's implement _those_ variants (and only those invariants) until such time as we require others.

Reply:
I've removed a few of the variants. What's left, based on a quick grep and analysis of our current use of 'internal' where we should be using other codes, as well as accepted RFCs, should be just what we need now or in the immediate future.